src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/EventBookingManager.java

```diff
@@ -734,29 +924,53 @@ public boolean hasBookingWithAnyOfStatuses(final String eventId, final Long user
      */
     public void cancelBooking(final IsaacEventPageDTO event, final RegisteredUserDTO user)
             throws SegueDatabaseException, ContentManagerException {
+
+        Long reservedById = null;
         try {
             // Obtain an exclusive database lock to lock the booking
             this.bookingPersistenceManager.acquireDistributedLock(event.getId());
-            this.bookingPersistenceManager.updateBookingStatus(event.getId(), user.getId(),
+            UserSummaryDTO reservedBy = this.bookingPersistenceManager
+                    .getBookingByEventIdAndUserId(event.getId(), user.getId()).getReservedBy();
+            reservedById = reservedBy == null ? null : reservedBy.getId();
+            BookingStatus previousBookingStatus = this.getBookingStatus(event.getId(), user.getId());
```

Comment:
This loads the same event booking at least twice, only to write back a new version of it. That seems unnecessary?

Reply:
Fixed in b7e7fe72.